modules/jupyter-widget/src/lib/components/add-support-components.js

```diff
@@ -0,0 +1,19 @@
+/* global document  */
+import getComponentByName from './get-component-by-name';
+
+export function addSupportComponents(container, props) {
+  const uiElementsOverlay = document.createElement('div');
+  uiElementsOverlay.className = 'deckgl-ui-elements-overlay';
+  uiElementsOverlay.style.zIndex = 1;
+  container.insertAdjacentElement('beforebegin', uiElementsOverlay);
+
+  const components = [];
+  for (const key of Object.keys(props)) {
```

Comment:
I'd much prefer if we listed out the props we support here in a switch statement rather than trying to make this so general - it is pretty hard to see what is going on right now.

```js
switch (key) {
case 'DescriptionCard': 
  ...
};
```

Also we don't normally have capitalized props. Coding it this way makes it easier to audit.